tests/test_quantity.py

```diff
@@ -17,13 +17,30 @@
     IncompatibleValue,
     InsufficientArguments,
     NumPyRequired,
+    RequiresUniqueDimensions,
     get_si_value,
 )
 from ansys.units.unit import IncorrectUnits
 
 DELTA = 1.0e-5
 
 
+def test_preferred_units():
+    Quantity.preferred_units(units=["J", "slug", "psi"])
+    assert Quantity._chosen_units == [Unit("J"), Unit("slug"), Unit("psi")]
+
```

Comment:
we would need a test where `preferred_units` is called again with same dimensions and different dimensions, checking that `_chosen_units` is updated appropriately.

Reply:
More tests have been added.